src/Matterhorn/State/Messages.hs

```diff
@@ -850,6 +873,31 @@ notifyGetPayload NotifyV3 st post mentioned = do
                 PostTypeChannelDeleted    -> "channelDeleted"
                 PostTypeEphemeral         -> "ephemeral"
                 PostTypeUnknown _         -> "unknown"
+notifyGetPayload NotifyV4 st post mentioned = do
+    let notification = NotificationV4 4 msg mentioned sender msgTy chName chType
+    return (A.encode notification)
+        where
+            chan = fromJust $ (findChannelById (post^.postChannelIdL) (st^.csChannels))
```

Comment:
This `fromJust` will raise a runtime exception if the channel is missing, which could happen due to a race. Instead, please move this into the `Maybe` monad above the `let` binding above, since this function can already fail gracefully with a `Nothing`:
```haskell
chan <- findChannelById (post^.postChannelIdL) (st^.csChannels)
```

Reply:
Oh, nice! Thanks for the suggestion! :)